Code/src/main/java/nl/utwente/group10/ui/components/Connection.java

```diff
@@ -206,4 +215,23 @@ public String toString() {
         return "Connection connecting \n(out) " + startAnchor + "   to\n(in)  "
                 + endAnchor;
     }
+    
+    /**
+     * This method evaluates the validity of the created connection.
+     * If the connection results in an invalid operation a visual
+     * error will be displayed.
+     */
+    public void checkError() {
+        if(startAnchor.isPresent() && endAnchor.isPresent()) {
+            try {
+                //TODO Obviously this will cause errors, we need a way to access the Env
+                endAnchor.get().getBlock().asExpr().analyze(new Env(), new GenSet());
+                this.setStyle("-fx-stroke: black;");
+            } catch (HaskellTypeError e) {
+                this.setStyle("-fx-stroke: red;");
+            } catch (HaskellException e) {
+                //e.printStackTrace();
```

Comment:
Hier zit ik dus stuck op, lees ook de TODO, we willen zeker in staat zijn om te kunnen kijken of een connectie geldig is, hier lijkt mij de analyze methode uitermate geschikt voor. Hoe benaderen we die correct? Hier moet veel code voor omgegooid worden lijkt mij maar ik kan er naast zitten.